span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -498,4 +523,13 @@ private void populateUrlParts(Request.Builder requestBuilder) {
       LOGGER.error("Error populating url parts", e);
     }
   }
+
+  private static boolean isAbsoluteUrl(String url) {
```

Comment:
As this is not very clearly testing absolute URL, we can have normalized function something like below
```
private URL getNormalizedUrl(String url) {
    return new URL(new URL("hypertrace.org"), url);
  }
```
And test above in populateUrlPars relativeness using string comparison too.

```
URL url = new URL(dummyUrl, urlStr);
      if (url.toString().equals(urlStr)) {
        requestBuilder.setScheme(url.getProtocol());
        requestBuilder.setHost(url.getAuthority()); // Use authority so in case the port is specified it adds it to this
      } else {    // relative URL
        requestBuilder.setUrl(null); //  unset the URL as we only allow absolute/full URLs in the url field
      }
```

Reply:
Made the suggested changes